### PR TITLE
[RFR] Added v2v tests

### PR DESCRIPTION
**Pull request by vatsalparekh**
New tests for V2V

### CODE REVIEW

File: tests/v2v/v2v_test.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+	"github.com/kubevirt/kubevirt-ansible/tests"
+	. "github.com/onsi/ginkgo"
+
+	ktests "kubevirt.io/kubevirt/tests"
+)
+
+var _ = Describe("Importing and Starting VMs via V2V", func() {
+	flag.Parse()
+	ktests.SkipIfNoOc()
+
```
Comment (nellyc) [inline tests/v2v/v2v_test.go]: we need some skip for the case there is no vmware instance to use

**Comment by nellyc** [comment]
ci test please

**Comment by vatsalparekh** [comment]
@nellyc @shiywang @gouyang PTAL
v2v tests would fail atm because of errors in v2v itself
but test does create serviceinstance so that's working

File: tests/v2v_test.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+const (
+	v2vServiceInstance    = "tests/manifests/import-vm-v2v-apb.yml"
+	dstv2vServiceInstance = "/tmp/test-v2v.json"
+)
+
+// Array of enviroment variables to get from testing env
+var listofEnvVars = []string{"ocpID", "ocpPass", "VMWareURL", "VMName", "VMWareID", "VMWarePass"}
+
+// Map to store Enviroment variables
+var v2vEnvVars map[string]string
```
Comment (shiywang) [inline tests/v2v_test.go]: I believe you can also use struct and iteratively fetch the value like this: http://www.golangprograms.com/how-to-get-struct-variable-information-using-reflect-package.html, But I'm okay with `map` as well.

File: tests/v2v_test.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+// Get all the Env variable needed for tests
+func getV2vEnv() {
+	v2vEnvVars = map[string]string{}
+	for _, vars := range listofEnvVars {
+		val, ok := os.LookupEnv(vars)
+		if !ok {
+			//raise exception
```
Comment (shiywang) [inline tests/v2v_test.go]: raise exception?

**Comment by shiywang** [comment]
others LGTM

File: tests/v2v_test.go (hunk 3 above)
Comment (vatsalparekh) [reply tests/v2v_test.go]: the env vars are important for the test, so `log.fatal` becomes important

File: tests/v2v_test.go (hunk 2 above)
Comment (vatsalparekh) [reply tests/v2v_test.go]: I feel `map` is fine here as storing them in a loop while taking input from array is easy and works

File: tests/manifests/import-vm-v2v-apb.yml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  spec:
+    clusterServiceClassExternalName: ${APB_NAME}
+    clusterServicePlanExternalName: vmware
+    parameters:
+      admin_user: ${OCP_ID}
+      admin_password: ${OCP_PASS}
+      url: ${VMWARE_URL}
+      vm_name: ${VM_NAME}
+      username: ${VMWARE_ID}
+      password: ${VMWARE_PASS}
```
Comment (pkliczewski) [inline tests/manifests/import-vm-v2v-apb.yml]: What about `os_type`? Do we want to have it here?

File: tests/manifests/import-vm-v2v-apb.yml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  annotations:
+    description: "Template for v2v import from VMWare"
+objects:
+- apiVersion: servicecatalog.k8s.io/v1beta1
+  kind: ServiceInstance
+  metadata:
+    name: import-vm-apb
+  spec:
+    clusterServiceClassExternalName: ${APB_NAME}
+    clusterServicePlanExternalName: vmware
```
Comment (pkliczewski) [inline tests/manifests/import-vm-v2v-apb.yml]: What about `vmware-template`? Do we want to verify it as well?

File: tests/manifests/import-vm-v2v-apb.yml (hunk 4 above)
Comment (vatsalparekh) [reply tests/manifests/import-vm-v2v-apb.yml]: included

File: tests/manifests/import-vm-v2v-apb.yml (hunk 5 above)
Comment (vatsalparekh) [reply tests/manifests/import-vm-v2v-apb.yml]: yes, included

File: tests/manifests/import-vm-v2v-apb.yml (hunk 4 above)
Comment (vatsalparekh) [reply tests/manifests/import-vm-v2v-apb.yml]: but it will only be useful when testing windows VMs

Comment (pkliczewski) [reply tests/manifests/import-vm-v2v-apb.yml]: correct

File: tests/v2v_test.go [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+var _ = Describe("Importing VMs using V2V", func() {
+
+	BeforeEach(func() {
+		getV2vEnv()
+	})
+
+	Context("Create ServiceInstance of import-vm-apb with vmware plan", func() {
+		It("should import vm suceessfully", func() {
+			tests.ProcessTemplateWithParameters(v2vServiceInstance, dstv2vServiceInstance,
```
Comment (shiywang) [inline tests/v2v_test.go]: 44-52 and 59-67 are exact same which can be reduced by either puting those code into a `BeforeEach` or `JustBeforeEach` block or using `table.Entry`

Comment (vatsalparekh) [reply tests/v2v_test.go]: https://github.com/kubevirt/kubevirt-ansible/blob/b5e629df47c2d733f54c426052f072eab07f4c4e/tests/v2v_test.go#L45
https://github.com/kubevirt/kubevirt-ansible/blob/b5e629df47c2d733f54c426052f072eab07f4c4e/tests/v2v_test.go#L60
@shiywang the param `APB_PLAN_NAME` is different in both, so it is not repeating same thing
Can you suggest a way to parametrize it?

Comment (shiywang) [reply tests/v2v_test.go]: can't you put those two into `v2vEnvVars`? if not another way is define a local var like `apbPlanName`, then both will be  `"APB_PLAN_NAME="+apbPlanName,`

Comment (vatsalparekh) [reply tests/v2v_test.go]: @shiywang @nellyc added use of Ginkgo table to parametrize tests from single Describe

File: tests/v2v_test.go (hunk 2 above)
Comment (lukas-bednar) [reply tests/v2v_test.go]: As we have discussed these config options will move to config file in future.

**Comment by vatsalparekh** [comment]
ci test please

**Pull request merged**